pkg/model/core/policy.go

```diff
@@ -0,0 +1,11 @@
+package core
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+type Policy interface {
+	GetNamespacedName() types.NamespacedName
+	GetTargetRef() *v1alpha2.PolicyTargetReference
```

Comment:
I'm okay with having this interface but not sure how we could benefit from it

Reply:
Actually my latter code change will use that:

```
func (r *resourceMapper) TargetGroupPolicyToService(ctx context.Context, tgp *v1alpha1.TargetGroupPolicy) *corev1.Service {
	if obj := r.policyToTargetRefObj(ctx, tgp, &corev1.Service{}); obj != nil {
		return obj.(*corev1.Service)
	} else {
		return nil
	}
}

func (r *resourceMapper) VpcAssociationPolicyToGateway(ctx context.Context, vap *v1alpha1.VpcAssociationPolicy) *gateway_api.Gateway {
	if obj := r.policyToTargetRefObj(ctx, vap, &gateway_api.Gateway{}); obj != nil {
		return obj.(*gateway_api.Gateway)
	} else {
		return nil
	}
}

func (r *resourceMapper) policyToTargetRefObj(ctx context.Context, policy core.Policy, retObj client.Object) client.Object { }
	
```